numpy/lib/arraysetops.py

```diff
@@ -380,106 +381,130 @@ def setxor1d(ar1, ar2, assume_unique=False):
+    array([0, 6])
     """
-    # Ravel both arrays, behavior for the first array could be different
-    ar1 = np.asarray(ar1).ravel()
-    ar2 = np.asarray(ar2).ravel()
+    # Ravel both arrays, will reshape result later to match elements
+    elements = np.asarray(elements)
+    elements_flat = elements.ravel()
+    test_elements = np.asarray(test_elements).ravel()
 
     # This code is significantly faster when the condition is satisfied.
-    if len(ar2) < 10 * len(ar1) ** 0.145:
+    if len(test_elements) < 10 * len(elements_flat) ** 0.145:
         if invert:
-            mask = np.ones(len(ar1), dtype=np.bool)
-            for a in ar2:
-                mask &= (ar1 != a)
+            mask = np.ones(len(elements_flat), dtype=np.bool)
+            for item in test_elements:
+                mask &= (elements_flat != item)
         else:
-            mask = np.zeros(len(ar1), dtype=np.bool)
-            for a in ar2:
-                mask |= (ar1 == a)
-        return mask
+            mask = np.zeros(len(elements_flat), dtype=np.bool)
+            for item in test_elements:
+                mask |= (elements_flat == item)
+        return mask.reshape(elements.shape)
 
     # Otherwise use sorting
     if not assume_unique:
-        ar1, rev_idx = np.unique(ar1, return_inverse=True)
-        ar2 = np.unique(ar2)
+        elements_flat, rev_idx = np.unique(elements_flat, return_inverse=True)
+        test_elements = np.unique(test_elements)
 
-    ar = np.concatenate((ar1, ar2))
+    elements_cat = np.concatenate((elements_flat, test_elements))
     # We need this to be a stable sort, so always use 'mergesort'
     # here. The values from the first array should always come before
     # the values from the second array.
-    order = ar.argsort(kind='mergesort')
-    sar = ar[order]
+    order = elements_cat.argsort(kind='mergesort')
+    elements_cat_sort = elements_cat[order]
     if invert:
-        bool_ar = (sar[1:] != sar[:-1])
+        bool_elements = (elements_cat_sort[1:] != elements_cat_sort[:-1])
     else:
-        bool_ar = (sar[1:] == sar[:-1])
-    flag = np.concatenate((bool_ar, [invert]))
-    ret = np.empty(ar.shape, dtype=bool)
+        bool_elements = (elements_cat_sort[1:] == elements_cat_sort[:-1])
+    flag = np.concatenate((bool_elements, [invert]))
+    ret = np.empty(elements_cat.shape, dtype=bool)
     ret[order] = flag
 
     if assume_unique:
-        return ret[:len(ar1)]
+        result = ret[:len(elements_flat)]
     else:
-        return ret[rev_idx]
+        result = ret[rev_idx]
+    return result.reshape(elements.shape)
+
+def in1d(elements, test_elements, **kwargs):
+    """
+    Test whether each element of a 1-D array is also present in a second array.
+    If `elements` is multidimensional, it is flattened.
+
+    Returns a boolean array of the same length as `elements` that is True
+    where an element of `elements` is in `test_elements` and False otherwise.
+
+    See `numpy.isin` for more details.
```

Comment:
I'm a little nervous about deleting all the parameters on this method, including the notes on when they were added. Can we keep them around, even though it's slightly redundant?